.github/workflows/ pr-workflow.yml

```
@@ -0,0 +1,154 @@
+          RUSTFLAGS: "-A warnings"
+          DATABASE_TYPE: postgres
+          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
+          BUCKET_NAME: mock
+          RUST_TEST_THREADS: 1
+        run: |
+          cd packages/api && make test
+
+  main-ui:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)